examples/resources/grafana_alert_rule/resource.tf

```diff
@@ -0,0 +1,85 @@
+resource "grafana_folder" "rule_folder" {
+    title = "My Alert Rule Folder"
+}
+
+resource "grafana_alert_rule" "my_alert_rule" {
+    name = "My Rule Group"
+    folder_uid = grafana_folder.rule_folder.uid
+    interval_seconds = 240
+    org_id = 1
+    rules {
+        name = "My Alert Rule 1"
+        for = 120
+        condition = "B"
+        no_data_state = "NoData"
+        exec_err_state = "Alerting"
+        annotations = {
+            "a" = "b"
+            "c" = "d"
+        }
+        labels = {
+            "e" = "f"
+            "g" = "h"
+        }
+        data {
+            ref_id = "A"
+            query_type = ""
+            relative_time_range {
+                from = 600
+                to = 0
+            }
+            datasource_uid = "PD8C576611E62080A"
+            model = jsonencode({
+                hide = false
+                intervalMs = 1000
+                maxDataPoints = 43200
+                refId = "A"
+            })
+        }
+        data {
+            ref_id = "B"
+            query_type = ""
+            relative_time_range {
+                from = 0
+                to = 0
+            }
+            datasource_uid = "-100"
+            model = <<EOT
```

Comment:
I designed this example to show 2 workflows. FIrst stage shows the HCL-based rule data, where the second stage shows how to do JSON-based rule data. Both are supported. First looks cleaner, second is nice if a user already has rule JSON ready to go.